sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_session_async.py

```diff
@@ -175,8 +177,14 @@ async def _incoming_attach(self, frame):
             self._input_handles[frame[1]] = self.links[frame[0].decode('utf-8')]  # name and handle
             await self._input_handles[frame[1]]._incoming_attach(frame)
         except KeyError:
-            outgoing_handle = self._get_next_output_handle()  # TODO: catch max-handles error
-            if frame[2] == Role.Sender:  # role
+            try:
+                outgoing_handle = self._get_next_output_handle()  # TODO: catch max-handles error
+            except ValueError:
+                self._connection.close(error=AMQPSessionError(
```

Comment:
This is async - so this needs to be awaited.
Additionally I'm not sure this should be a connection close - it's currently only the Session that's invalid?
Lastly - AMQPSessionError is an exception - so you can't pass it into connection.close.

Reply:
I went with `close` after reading the `handle-max` description [here](http://docs.oasis-open.org/amqp/core/v1.0/csprd01/amqp-core-transport-v1.0-csprd01.html#type-begin). 

>  A peer that receives a handle outside the supported range MUST close the connection with the framing-error error-code


Would this only apply to the begin frame? Seeing a couple of other libraries its just an exception that is raised, so it could be I'm reading it incorrectly ( client generating an out of range id versus peer receiving one)

Reply:
Receiving a handle outside the supported range is not the same as running out of available handles.
If would mean that `frame[1]` was a value > `handle-max`.
That's not to say that closing the connection is invalid - however we have to assume that this may not be the only open session, and if we've run out of handles, there's probably additional open links. So calling `connection.close()` must gracefully shut down all other sessions and links on the connection.

However rather than close the session it might be more valid to detach the attempted link?

Lastly, this still needs an await, and you cannot pass in AMQPSessionError, it must be AMQPError.

Reply:
I went over uamqp again and on an invalid handle_max the link is freed up, so detaching 
 the link would be the valid choice.

Reply:
Updated this to detach the link that would have been assigned and then return